php/index.php

```diff
@@ -0,0 +1,108 @@
+$og_title = "NSG Comprehensive Rules";
+$og_description = "Description";
+
+if(isset($_GET['r'])) {
+    $rule = $_GET['r'];
+} else {
+    $rule = '';
+}
+
+$path = 'json/rules.json';
+$jsonString = file_get_contents($path);
+$jsonData = json_decode($jsonString, true);
+
+function search_for_rule($jsonData, $key, $value) {
+    foreach($jsonData as $r){
+        if($r[$key] == $value) {
+            return $r; 
+        }
+    }
+}
+
+$og_desc = "Access the complete rules framework that makes Netrunner click.";
+$jsonRule = null;
+if($rule) {
+    $jsonRule = search_for_rule($jsonData, 'id', $rule);
+}
+
+if($jsonRule) {
+    /* Three different scenarios:
+        $jsonRule is level 1 (i.e. chapter):
+            - og:title will be $jsonRule
+            - og:description will be empty
+        $jsonRule is level 2 (i.e. subchapter):
+            - og:title will be $jsonRule
+            - og:description will be its children
+        $jsonRule is level 3 or deeper (i.e. subrule):
+            - og:title will be the parent section
+            - og:description will be $jsonRule and its children
+    */
+    $level = count(explode('.', $jsonRule['nr']));
+
+    if($level == 1) {
+        $og_title = $jsonRule['nr'].". ".$jsonRule['text'];
+        $og_desc = "";
+    } else {
+        if($level == 2) {
+            $og_title = $jsonRule['nr'].". ".$jsonRule['text'];
+            $skipSelf = true;
+        }
+        if($level > 2) {
+            // Look for parent section and display it as title
+            $sectionNR = join('.',array_slice(explode('.', $jsonRule['nr']), 0, -1));
+            $jsonSection = search_for_rule($jsonData, 'nr', $sectionNR);
+            
+            $og_title = $jsonSection['nr'].". ".$jsonSection['text'];
+            $skipSelf = false;
+        }
+
+        // Fill text with this item and its children
+        $out = "";
+        function appendItem(&$out, &$jsonData, $jsonElem, $skipSelf){
+            if(strlen($out) > 500) {
+                return;
+            }
+            if(!$skipSelf){
+                $out = $out." ".$jsonElem['nr'].". ".$jsonElem['text'];
+            }
+            foreach($jsonElem['children'] as $id) {
+                $childElem = search_for_rule($jsonData, 'id', $id);
+                if($childElem) {
+                    appendItem($out, $jsonData, $childElem, false);
+                }
+            }
+        }
+        appendItem($out, $jsonData, $jsonRule, $skipSelf);
+
+        $og_desc = trim($out);
+    }
+}
+
+$doc = new DOMDocument('1.0', 'UTF-8');
+$internalErrors = libxml_use_internal_errors(true);
+$doc->loadHTMLFile("html/rules.html");
+
+$head = $doc->getElementsByTagName('head')[0];
+
+$metaProps = array();
+$metaProps[] = array("property" => "og:title", "content" => $og_title);
+$metaProps[] = array("property" => "og:type", "content" => "website");
+$metaProps[] = array("property" => "og:image", "content" => "https://dumpyard.lostgeek.de/rules/logo.png");
```

Comment:
is there a way to make this relative to the current site?

Reply:
Unfortunately not, as far as I can see. The [OpenGraph protocol](https://ogp.me/) needs a full URL there. And according to [this StackOverflow post](https://stackoverflow.com/questions/10717249/get-current-domain/50301646#50301646) the only secure way of getting the current domain name in php is to save it to a config file yourself.

Reply:
I did it via config file. The python script now has 2 additional parameters for enabling php output and defining the server base path:
```
usage: __main__.py [-h] [-a] [-y YEAR] [-m MONTH] [-d DAY] [-p] [-b PHP_BASE_PATH]

options:
[...]
  -p, --generate-php    Generate php output
  -b PHP_BASE_PATH, --php-base-path PHP_BASE_PATH
                        Basepath of php server
 ```